teuthology/orchestra/connection.py

```diff
@@ -29,14 +30,12 @@ def create_key(keytype, key):
     """
     Create an ssh-rsa, ssh-dss or ssh-ed25519 key.
     """
-    if keytype == 'ssh-rsa':
-        return paramiko.rsakey.RSAKey(data=base64.decodestring(key.encode()))
-    elif keytype == 'ssh-dss':
-        return paramiko.dsskey.DSSKey(data=base64.decodestring(key.encode()))
-    elif keytype == 'ssh-ed25519':
-        return paramiko.ed25519key.Ed25519Key(data=base64.decodestring(key.encode()))
-    else:
-        raise ValueError('keytype must be ssh-rsa, ssh-dss (DSA) or ssh-ed25519')
+    l = "{hostname} {keytype} {key}".format(hostname="x", keytype=keytype, key=key)
+
+    ke = HostKeyEntry.from_line(l)
+    assert ke,
+        'invalid host keytype {keytype}'.format(keytype = keytype))
```

Comment:
this changes the behavior of `create_key()`, i'd prefer having this change in a separate commit. but since we are already using `assert` for error handling in `split_user()`. i am fine with it. how about printing out the whole line, as, i assume a malformed key could also contribute the failure, like
```py
assert ke, f'invalid host key "{keytype} {key}"'
```
would suffice.